src/server/pfs/server/driver.go

```diff
@@ -2518,7 +2603,14 @@ func (d *driver) applyWrite(key string, records *pfs.PutFileRecords, tree hashtr
 			indexOffset++ // start writing to the file after the last file
 		}
 		for i, record := range records.Records {
-			if err := tree.PutFile(path.Join(key, fmt.Sprintf(splitSuffixFmt, i+int(indexOffset))), []*pfs.Object{{Hash: record.ObjectHash}}, record.SizeBytes); err != nil {
+			if err := tree.PutFileSplit(
+				path.Join(key, fmt.Sprintf(splitSuffixFmt, i+int(indexOffset))),
+				[]*pfs.Object{{Hash: record.ObjectHash}},
+				record.SizeBytes,
+				&pfs.Object{Hash: records.Header.ObjectHash},
+				&pfs.Object{Hash: records.Footer.ObjectHash},
+				records.Header.SizeBytes+records.Footer.SizeBytes,
+			); err != nil {
```

Comment:
So is it accurate that we're passing the Header and Footer once for each datum we write out? That's probably not a big deal performance wise but it is a little inefficient and I think it might make more sense to instead have a special call, like maybe `tree.PutHeader` and `tree.PutFooter` so you can explicitly put the correct values into the directory once rather than doing it implicitly with each file you want to put.

Reply:
Made this change